File: src/sanitize-string-filename.js
import sanitizeStringWord from './sanitize-string-word';

/**
 * @param {string} fileName - The given file name string to be sanitized.
 * @returns {string} The given file name with all invalid characters removed.
 */
const sanitizeStringFilename = (fileName) => {
  const extension = fileName.split('.').pop();

  return `${sanitizeStringWord(fileName.replace(extension, ''))}.${sanitizeStringWord(extension)}`;
};

export default sanitizeStringFilename;

File: sanitize_string_filename.go
package main

import (
	"fmt"
	"strings"
)

// Gets the given file name with all invalid characters removed.
func SanitizeStringFilename(value string) string {
	var parts []string = strings.Split(value, ".")

	var extension string = parts[len(parts)-1]

	// FIXME: allow "/"
	return fmt.Sprintf("%v.%v", SanitizeStringWord(strings.Replace(value, extension, "", -1)), SanitizeStringWord(extension))
}
